drivers/sensor/weplatform/weplatform.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2022 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "weplatform.h"
+
+#ifdef CONFIG_I2C
+#include <drivers/i2c.h>
+#endif /* CONFIG_I2C */
+
+#ifdef CONFIG_SPI
+#include <drivers/spi.h>
+#endif /* CONFIG_SPI */
+
+/**
+ * @brief Read data starting from the addressed register
+ * @param[in] interface Sensor interface
+ * @param[in] regAdr The register address to read from
+ * @param[in] numBytesToRead Number of bytes to read
+ * @param[out] data The read data will be stored here
+ * @retval Error code
+ */
+inline int8_t WE_ReadReg(WE_sensorInterface_t *interface, uint8_t regAdr, uint16_t numBytesToRead,
+			 uint8_t *data)
+{
+	int status = 0;
+
+	switch (interface->interfaceType) {
+	case WE_i2c:
+#ifdef CONFIG_I2C
+		if (numBytesToRead > 1 && interface->options.i2c.useRegAddrMsbForMultiBytesRead) {
+			/*
+			 * Register address most significant bit is used to
+			 * enable multi bytes read
+			 */
+			regAdr |= 1 << 7;
```

Comment:
[WRITE_BIT](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util_macro.h#L61)

Reply:
Done (and moved to hal_we module).